yt_dlp/extractor/novaplay.py

```diff
@@ -6,44 +6,49 @@ class NovaPlayIE(InfoExtractor):
     _VALID_URL = r'https://play.nova\.bg/video/.*/(?P<id>\d+)'
     _TESTS = [
         {
-            'url': 'https://play.nova.bg/video/bratya/season-3/bratq-2021-10-08/548677',
-            'md5': 'b1127a84e61bed1632b7c2ca9cbb4153',
+            'url': 'https://play.nova.bg/video/ochakvaite/season-0/ochakvaite-2022-07-22-sybudi-se-sat/606627',
+            'md5': 'd79dff2d09d196c595a7290f48e33399',
             'info_dict': {
-                'id': '548677',
+                'id': '606627',
                 'ext': 'mp4',
-                'title': 'Братя',
-                'alt_title': 'bratya/season-3/bratq-2021-10-08',
-                'duration': 1603.0,
-                'timestamp': 1633724150,
-                'upload_date': '20211008',
-                'thumbnail': 'https://nbg-img.fite.tv/img/548677_460x260.jpg',
-                'description': 'Сезон 3 Епизод 25'
+                'title': 'Събуди се - събота по NOVA (23.07.2022)',
+                'alt_title': 'ochakvaite/season-0/ochakvaite-2022-07-22-sybudi-se-sat',
+                'duration': 29.0,
+                'timestamp': 1658491547,
+                'upload_date': '20220722',
+                'thumbnail': 'https://nbg-img.fite.tv/img/606627_460x260.jpg',
+                'description': '29 сек',
+                'view_count': False
             },
         },
         {
-            'url': 'https://play.nova.bg/video/igri-na-volqta/season-3/igri-na-volqta-2021-09-20-1/548227',
-            'md5': '5fd61b8ecbe582fc021019d570965d58',
+            'url': 'https://play.nova.bg/video/ochakvaite/season-0/ochakvaite-2022-07-22-cherry-tazi/606609',
+            'md5': 'f3e973e2ed1a5b9b3f498b1ab82d01b3',
             'info_dict': {
-                'id': '548227',
+                'id': '606609',
                 'ext': 'mp4',
-                'title': 'Игри на волята: България (20.09.2021) - част 1',
-                'alt_title': 'gri-na-volqta/season-3/igri-na-volqta-2021-09-20-1',
-                'duration': 4060.0,
-                'timestamp': 1632167564,
-                'upload_date': '20210920',
-                'thumbnail': 'https://nbg-img.fite.tv/img/548227_460x260.jpg',
-                'description': 'Сезон 3 Епизод 13'
+                'title': 'Черешката на тортата - тази вечер по NOVA (22.07.2022)',
+                'alt_title': 'ochakvaite/season-0/ochakvaite-2022-07-22-cherry-tazi',
+                'duration': 29.0,
+                'timestamp': 1658476303,
+                'upload_date': '20220722',
+                'thumbnail': 'https://nbg-img.fite.tv/img/606609_460x260.jpg',
+                'description': '29 сек',
+                'view_count': False
             },
         }
     ]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
         webpage = self._download_webpage(url, video_id)
+        access_token = self._download_json('https://play.nova.bg/api/client', None)['accessToken']
```

Comment:
Move to _real_initialize

Reply:
this depends on the initial webpage download going by the earlier version so don't think that would work.

OP should add a comment about it

Reply:
Is the token URL dependent? If not, downloading a random page (home page?) in real_initialize would be better imo. Current implementation adds too many requests for batch downloads.

Or if you really don't want to prevent that additional request, this can be kept in real_extract; but save the token in `self`. Like:
```py
if not self._access_token:
    self._access_token = ...
```

Reply:
I need to download a page (does't matter which one) to get the cookies needed for getting the access token. Because I download the page for a specific video anyways I decided to initialize the access_token in real_initialize and then get it the first time the page for a video is downloaded.